plugin/pkg/scheduler/factory/factory.go

```diff
@@ -338,9 +351,28 @@ func (c *ConfigFactory) onPvcDelete(obj interface{}) {
 }
 
 func (c *ConfigFactory) invalidatePredicatesForPvc(pvc *v1.PersistentVolumeClaim) {
-	if pvc.Spec.VolumeName != "" {
-		c.equivalencePodCache.InvalidateCachedPredicateItemOfAllNodes(maxPDVolumeCountPredicateSet)
+	// We need to do this here because the ecache uses PVC uid as part of equivalence hash of pod
+
+	// The bound volume type may change
+	invalidPredicates := sets.NewString(maxPDVolumeCountPredicateKeys...)
+
+	// The bound volume's label may change
+	invalidPredicates.Insert("NoVolumeZoneConflict")
```

Comment:
You also need to invalidate the NoVolumeNodeConflict predicate under the LocalPersistentVolumes feature gate for PVC delete.

Reply:
Thanks, just fixed.